File: javalin-openapi-processor/src/main/java/net/dzikoysk/openapi/processor/annotations/OpenApiFormParamInstance.java
package net.dzikoysk.openapi.processor.annotations;

import io.javalin.plugin.openapi.annotations.OpenApiFormParam;
import net.dzikoysk.openapi.processor.processing.AnnotationMirrorMapper;

import javax.lang.model.element.AnnotationMirror;
import java.lang.annotation.Annotation;

@SuppressWarnings("ClassExplicitlyAnnotation")
final class OpenApiFormParamInstance extends AnnotationMirrorMapper implements OpenApiFormParam {

    OpenApiFormParamInstance(AnnotationMirror mirror) {
        super(mirror);
    }

    @Override
    public String name() {
        return getString("name");
    }

    @Override
    public boolean required() {
        return getBoolean("required");
    }

    @Override
    public Class<?> type() {
        return null;
    }

    @Override
    public Class<? extends Annotation> annotationType() {
        return OpenApiFormParam.class;
    }

}

File: openapi-processor/src/main/kotlin/net/dzikoysk/openapi/processor/annotations/OpenApiFormParamInstance.kt
package net.dzikoysk.openapi.processor.annotations

import net.dzikoysk.openapi.annotations.OpenApiFormParam
import net.dzikoysk.openapi.processor.processing.AnnotationMirrorMapper
import javax.lang.model.element.AnnotationMirror
import javax.lang.model.type.TypeMirror

internal class OpenApiFormParamInstance internal constructor(mirror: AnnotationMirror) : AnnotationMirrorMapper(mirror) {

    fun name(): String =
        getString("name")

    fun required(): Boolean =
        getBoolean("required")

    fun type(): TypeMirror =
        getType("type")

    fun annotationType(): Class<out Annotation> =
        OpenApiFormParam::class.java

}